src/it/scala/djinni/IntegrationTest.scala

```diff
@@ -193,12 +213,57 @@ class IntegrationTest extends AnyFunSpec {
         Source.fromFile(s"src/it/resources/result/$idl/$lang/$filename")
       val expectationFile =
         Source.fromFile(s"src/it/resources/expected/$idl/$lang/$filename")
-      resultFile.mkString should equal(expectationFile.mkString)
+      val resultFileLines = resultFile.getLines
+      val expectationFileLines = expectationFile.getLines
+      for (
+        (result, expectation) <- resultFileLines.zipAll(
+          expectationFileLines,
+          "",
+          ""
+        )
+      ) {
+        result should equal(expectation)
+      }
+
       resultFile.close()
       expectationFile.close()
     }
   }
 
+  /** Asserts that all generated files are the same. It basically compares the
+    * paths listed in generated files
+    * `resources/result/$idl/$lang/generated-files.txt` with the expectations
+    * defined in `resources/expected/$idl/$lang/generated-files.txt`.
+    * @param idl
+    *   filename of the input-idl (without file extension)
+    * @param lang
+    *   language to assert for (e.g. `cpp`, `cpp-headers`, `java`, `jni`,
+    *   `jni-headers`, `objc`, `objc-headers`, `objcpp`, `objcpp-headers`)
```

Comment:
Copy and paste error? There is no `lang` parameter in `assertGeneratedFilesEqual`.

```suggestion
```

Reply:
I'll remove it when I push reorganized commits; I would like to wait until we resolve above conversations :)